google/cloud/pubsublite/internal/topic_partition_count_reader.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_TOPIC_PARTITION_COUNT_READER_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_INTERNAL_TOPIC_PARTITION_COUNT_READER_H
+
+#include "google/cloud/pubsublite/topic.h"
+#include "google/cloud/future.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/version.h"
+#include <utility>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace pubsublite_internal {
+
+/**
+ * This interface declares an asynchronous method to get the number of
+ * partitions for a Pub/Sub Lite topic.
+ */
+class TopicPartitionCountReader {
+ public:
+  virtual future<StatusOr<std::uint64_t>> Read(
```

Comment:
nit. make this `std::int64_t` to match the proto value type

Reply:
Alternatively, make this `std::uint32_t` to match the needed bounds and check that the value is within those bounds.